libzeth/circuits/blake2s/blake2s.hpp

```diff
@@ -25,66 +25,67 @@ template<typename FieldT> class BLAKE2s_256 : public libsnark::gadget<FieldT>
 {
 private:
     // Parameter block, size set to 32 bytes, fanout and depth set to serial
-    // mode. See: Section 2.8 https://blake2.net/blake2.pdf Table 2
-    std::array<std::array<bool, BLAKE2s_word_size>, 8> parameter_block = {
+    // mode. See: Section 2.8 https://blake2.net/blake2.pdf Table 2. Note, byte
+    // endianness has been swapped to avoid doing this repeatedly at runtime.
```

Comment:
What do you mean by "to avoid doing this repeatedly at runtime"?

Reply:
Previously, these constants were endian-swappd before being used (see the .tcc file), and this swap was happening for every invocation of `generate_r1cs_witness` for every instance of the gadget.  Those constants did not seem to be used anywhere else, so the byte swapping seemed unnecessary.

> What do you mean by "to avoid doing this repeatedly at runtime"?

It refers to the byte swapping.  It's done in the static data, so it doesn't have to be done many times when running the proof gen.  (I put the comment to for anyone looking at the reference above).

Reply:
Ah yes, I see. The parameter block structure was taken from Table 2. https://blake2.net/blake2.pdf and since blake processed byte strings as little endian the endianness needed to be switched.
Yes that's fine to switch the endianness directly in the parameter block definition. This is what is done in the RFC (see section 2.5 https://tools.ietf.org/html/rfc7693#page-5)